npm/src/main.ts

```diff
@@ -96,9 +115,17 @@ export function getDebugServiceWorker(): IDebugServiceWorker {
   return proxy;
 }
 
-export function getLanguageService(): ILanguageService {
+export function getLanguageService(
+  readFile?: (uri: string) => Promise<string | null>,
+  listDir?: (uri: string) => Promise<[string, number][]>,
+  getManifest?: (uri: string) => Promise<{
+    excludeFiles: string[];
+    excludeRegexes: string[];
+    manifestDirectory: string;
+  } | null>,
+): ILanguageService {
```

Comment:
Just to make sure I understand, the expected behavior when `getLanguageService()` is called without passing any of these parameters is that the callbacks are null and it's the responsibility of the caller to not use any of the file system APIs? From what I can tell, there are no checks after this point so the language service is just holding those null pointers and may try to call them if the file APIs are used when the langauge service hasn't been initialized with callbacks, resulting in a null dereference crash. Not sure if it's worth the extra effort to add checks to the Rust internals that panic with a message before trying to invoke callbacks versus letting it crash and forcing a dev to connect a debugger... it's a development scenario either way, so it's really a question of how easy we want it to be for developers to diagnose the crash when they are using the npm module improperly.